### ISSUE TITLE
[Bug]: Styled returns "Functions are not valid as a React child"

### ISSUE THREAD

**Issue by jmeistrich** [marklawlor/tailwindcss-react-native#42]
I can't seem to get `styled` or `StyledComponent` to work. They crash with this error in styled.js:

`Warning: Functions are not valid as a React child. This may happen if you return a Component instead of <Component /> from render. Or maybe you meant to call this function rather than return it.`

I tried this in my test Next.js project, with a repro here:  https://github.com/jmeistrich/tailwindcss-react-native-next

I also tried following your install instructions in a fresh Expo app, doing the quickstart (https://tailwindcss-react-native.vercel.app/quick-start) then https://tailwindcss-react-native.vercel.app/api/styled. And it crashes with the same error.

**Comment by marklawlor** [comment]
This occurs when both the Components API and the Babel transform is being using (resulting double wrapped styled components) e.g

```
<StyleComponent component={<StyledCompoent component={<YourComponent />} />} />
```

If you want to use the Component API with babel you can set the `mode` to `compileOnly` https://tailwindcss-react-native.vercel.app/configuration/babel.

I'll keep this open until I come up with a better method to warn about usage.

**Comment by jmeistrich** [comment]
Ah ok. But there may be cases where they need to work together? For example if I have allowModuleTransform set to a local components folder, but then I import a module that I want to create a styled version for. In that case I think I would need to wrap the module's components in `styled(...)`? It seems like even if babel is not transforming an imported component it's still crashing with this error. Or how would you suggest doing that?

I am looking to add a set of styled components to my library https://github.com/LegendApp/legend-motion as a convenience, but then would they break if the consumer is using the babel transform?

It seems like the alternative is to require the user to add `allowModuleTransform: ['@legendapp/motion']` so the babel plugin will transform them?

**Comment by marklawlor** [comment]
Yep I'm keeping this open for that reason, I'll have a fix for it soon.

**Comment by marklawlor** [comment]
Fixed in `1.0.4`
